tsdb/meta.go

```diff
@@ -139,11 +139,11 @@ func (d *DatabaseIndex) CreateSeriesIndexIfNotExists(measurementName string, ser
 
 	series.measurement = m
 	d.series[series.Key] = series
-	d.mu.Unlock()
 
 	m.AddSeries(series)
 
 	d.statMap.Add(statDatabaseSeries, 1)
+	d.mu.Unlock()
```

Comment:
Should be. I'm just reverting so the behavior matches https://github.com/influxdata/influxdb/commit/ff1447202c402196f52ffd2eb4f1a3a4d9f1aa65#diff-747cff278618796cefbbc702a1dca575L136 as before though.